analyzers/src/SonarAnalyzer.CSharp/Rules/InterfacesShouldNotBeEmpty.cs

```diff
@@ -42,14 +42,29 @@ public sealed class InterfacesShouldNotBeEmpty : SonarDiagnosticAnalyzer
 
                     var interfaceSymbol = c.SemanticModel.GetDeclaredSymbol(interfaceDeclaration);
                     if (interfaceSymbol is { DeclaredAccessibility: Accessibility.Public }
-                        && !IsAggregatingOtherInterfaces(interfaceSymbol))
+                        && !IsAggregatingOtherInterfaces(interfaceSymbol)
+                        && !IsSpecializedGeneric(interfaceSymbol)
+                        && !HasEnhancingAttribute(interfaceSymbol))
                     {
                         c.ReportIssue(Diagnostic.Create(Rule, interfaceDeclaration.Identifier.GetLocation()));
                     }
                 },
                 SyntaxKind.InterfaceDeclaration);
-
         private static bool IsAggregatingOtherInterfaces(ITypeSymbol interfaceSymbol) =>
             interfaceSymbol.AllInterfaces.Length > 1;
```

Comment:
Educational/suggestion: I think we shouldn't include interfaces that are base interfaces of directly implemented interfaces
```suggestion
            interfaceSymbol.Interfaces.Length > 1;
```

Reply:
Reproducer: 965983b1dcd099a81ffcfd4a32779a3f09d2140c (test red)
Fix: f3534cfcd55d5ecdef1d9c02dcd1c679705bdb60